### PR TITLE
Fix indexing bugs in geostrophy.distance()

### PR DESCRIPTION
**Pull request by efiring**

### CODE REVIEW

**Comment by efiring** [comment]
This passes the test locally now.  If you would like to add more tests, that would be good.

**Comment by efiring** [comment]
The py.test framework seems to be getting confused when reporting where the error came from; the detailed info at the end of the log is showing the wrong functions.

**Comment by ocefpaf** [comment]
> The py.test framework seems to be getting confused when reporting where the error came from; the detailed info at the end of the log is showing the wrong functions.

Are you refering to the warnings summary from line [814](https://travis-ci.org/TEOS-10/GSW-Python/jobs/242251484#L814)? That is mostly noise and we should probably try to supress that.
The actual tests seems to be OK and printing the right function name, no?

File: gsw/geostrophy.py [hunk 1]
```diff
@@ -155,17 +155,27 @@ def distance(lon, lat, p=0, axis=-1):
...
                           % (lon.shape, axis))
+    if lon.ndim == 1:
+        one_d = True
+        lon = lon[np.newaxis, :]
+        lat = lat[np.newaxis, :]
+        axis = -1
+    else:
+        one_d = False
+
+    one_d = one_d and p.ndim == 1
```
Comment (ocefpaf) [inline gsw/geostrophy.py]: More elegant than the solution I had in mind. Thanks!

**Pull request merged**